packages/lexical-list/src/LexicalListItemNode.ts

```diff
@@ -81,20 +80,19 @@ export class ListItemNode extends ElementNode {
     $setListItemThemeClassNames(element, config.theme, this);
     return element;
   }
+  updateDOM(prevNode: this, dom: HTMLElement, config: EditorConfig): boolean {
+    if (super.updateDOM(prevNode, dom, config)) {
+      return true;
+    }
 
-  updateDOM(
-    prevNode: ListItemNode,
-    dom: HTMLElement,
-    config: EditorConfig,
-  ): boolean {
     const parent = this.getParent();
     if ($isListNode(parent) && parent.getListType() === 'check') {
-      updateListItemChecked(dom, this, prevNode, parent);
+      const listItemNode = $isListItemNode(prevNode) ? prevNode : null;
+      updateListItemChecked(dom, this, listItemNode, parent);
```

Comment:
This change is not necessary, the type is already `this` which is known to be a `ListItemNode` (or subclass)

```suggestion
      updateListItemChecked(dom, this, prevNode, parent);
```